core/templates/domain/topic/subtopic.model.ts

```diff
@@ -13,22 +13,24 @@
 // limitations under the License.
 
 /**
- * @fileoverview Factory for creating and mutating instances of frontend
+ * @fileoverview Model for creating and mutating instances of frontend
  * subtopic domain objects.
  */
 
-import { ShortSkillSummary } from
-  'domain/skill/short-skill-summary.model';
+import { ShortSkillSummary } from 'domain/skill/short-skill-summary.model';
 
 import constants from 'assets/constants';
 
 export interface SubtopicBackendDict {
   'id': number;
   'title': string;
   'skill_ids': string[];
-  'thumbnail_filename': string;
-  'thumbnail_bg_color': string;
-  'url_fragment': string;
+  // 'null' value for 'thumbnail_filename', 'thumbnail_bg_color' &
```

Comment:
Same note as above -- null should be reserved for "this is meant to be None" rather than being a temporary thing. It defeats the point of types a bit if we type something one way with plans to change it later.

Reply:
But how can a newly created subtopic have empty values for these properties? When creating a subtopic from title these properties should be null initially see function`createFromTitle`.

Reply:
Can you show me a video of the UI where a subtopic can be created by providing only a title? I thought subtopic creation was gated on filling in all the fields.

If this is the case then we should have the code reflect that. For example, if subtopic addition is adding a "temporary incomplete subtopic" then that is messy and we shouldn't do that; instead, we should store the variables locally in the modal and only create a subtopic object on-save. Type annotation is intended to root out and fix these cases.

Reply:
Yes we have to fill all the fields but only for the frontend, populating a dummy topic creates a subtopic with just title property and we can even publish that 

https://user-images.githubusercontent.com/60519303/121150275-fe12e980-c860-11eb-9e1f-eb9445ca6631.mp4

Reply:
That seems bad, I think that needs to be fixed. How about refactoring so that we instead create a local subtopic object in the modal like I suggested in the second paragraph of my previous comment? As can be seen in the video, the current behaviour is suboptimal.

Reply:
@seanlip I'm not sure what you meant here ? function `createFromTitle` is used rightly here ,after all it is dependent on `create` which takes data from the user and then only creates a subtopic. This thing is only possible with admin priviliges (dependent on the backend) and should be right enough for testing purposes.

Reply:
@seanlip According to me function name is incorrect here,please note:- we're not doing this thing shown in the video only using frontend. For reference see this heirarchy: - `createFromTitle ` => `addSubtopic` from topic-update-service => this is used in `create-new-subtopic-modal.controller,ts`.In addition to it there is a check in the controller itself that we have to pass every value i.e title, url-fragment, description and a thumbnail for creating a subtopic.A user by any chance on production cannot create a subtopic with just a title property.

Reply:
Got that will be covering in a separate PR!

Reply:
Update: chatted with @mridul-netizen to explain this a bit more and I think we're on the same page now!

(@mridul-netizen feel free to reply to this comment saying otherwise, if that's not the case.)

Reply:
You'll need to drop the changes here before this PR can be approved (if you're planning to do the proper fix in a separate PR).

Reply:
done